include/spdlog/details/registry-inl.h

```diff
@@ -84,6 +85,22 @@ SPDLOG_INLINE std::shared_ptr<logger> registry::get(const std::string &logger_na
     return found == loggers_.end() ? nullptr : found->second;
 }
 
+#if __cplusplus >= 201703L  // C++17
+SPDLOG_INLINE std::shared_ptr<logger> registry::get(std::string_view logger_name) {
+    std::lock_guard<std::mutex> lock(logger_map_mutex_);
+    using key_value_pair = std::pair<std::string, std::shared_ptr<spdlog::logger>>;
+    const auto match = [logger_name](const key_value_pair &item) {
+        return item.first == logger_name;
+    };
+    const auto found = std::find_if(loggers_.cbegin(), loggers_.cend(), match);
+    return found == loggers_.end() ? nullptr : found->second;
+}
+
```

Comment:
since this is cpp17, a much shorter version would be

```

std::lock_guard<std::mutex> lock(logger_map_mutex_);
for (auto const& [key, val] : loggers_) {
  if(logger_name == key) return val;
}
return nullptr;
```

And then no need to `#include <algorithm>`